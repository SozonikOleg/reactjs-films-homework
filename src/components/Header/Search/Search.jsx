import React, { Component } from 'react';
import styles from './Search.module.scss';


class Search extends Component {
  render () {
    return (
      <div className={styles.three}>
        <h1> Search </h1>    
      </div>
    );
  }
} 

export default Search;