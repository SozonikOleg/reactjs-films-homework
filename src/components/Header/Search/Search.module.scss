.three {
    margin-left: 80%;
    color: blue;
    border: 1px solid blue;
}
