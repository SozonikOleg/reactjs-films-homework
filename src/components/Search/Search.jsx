import React, { Component } from 'react';
import { connect } from 'react-redux';
import styles from './Search.module.scss';
import { getSearchData } from '../../action/search';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

class Search extends Component {
  constructor(prop) {
    super(prop);
    this.state = {
      searchValue: '',
    };
  }

  componentDidMount() {
    this.clearForm();
  }

  hadleChangeInput = (event) => {
    this.setState({
      searchValue: event.target.value,
    });
  };

  searchMovies = (e) => {
    this.props.postSearch(this.state.searchValue);
    this.clearForm();
  }

  clearForm = () => {
    document.getElementById("myForm").reset(); 
    this.setState({
      searchValue: ''
    })
  }

  render() {
    const valueSearchRouting = this.state.searchValue;
    console.log('wdwdwdwdwd',this.state.searchValue);
    console.log('_______',this.props);
    return (
      <Router>
        <div className={styles.three1}>
          <section className={styles.logo}>
            <h2> Film</h2>
          </section>
          <section className={styles.form_search}>
            <form id="myForm">
              <div className={styles.search_wrapper}>
                <input
                  type="search"
                  placeholder="the jungle book"
                  value={this.state.searchValue}
                  onChange={this.hadleChangeInput}
                />
                <Link to={{ pathname: "/SearchResultsPage", search: `?search=${this.state.searchValue}`,}}>
                  <button className={styles.search_submit} type="submit" onClick={this.searchMovies.bind(this)} />
                </Link>
              </div>
            </form>
          </section>
        </div>
      </Router>
    );
  }
}

export default connect(
  state => ({}),
  dispatch => ({
    postSearch: SearchValue => {
      dispatch(getSearchData(SearchValue));
    },
  }),
)(Search);
