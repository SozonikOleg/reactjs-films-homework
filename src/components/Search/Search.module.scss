.three1 {
	display: flex;
	justify-content: space-between;
	padding-top: 21px;
}

.logo{
	margin-left: 101px;
	font-size:25px;
	text-transform: uppercase;
	color:white;
}

.form_search {
	margin-right: 90px; 
}

input:focus, input:active {
    outline: none !important;
}

input {
	outline: none;
}

input[type=search] {
	-webkit-appearance: textfield;
	-webkit-box-sizing: content-box;
	font-family: inherit;
	font-size: 100%;
	color: white;
}

.search_submit {
	width: 30px;
	height: 30px;
	background:url('../../image/search.png') no-repeat center;
	border: 0px;
	padding: 12px 24px;
}

.search_wrapper {
	border: solid 3px white;	
	-webkit-border-radius: 8px;
	-moz-border-radius: 8px;
	border-radius: 8px;
}

input::-webkit-search-decoration,
input::-webkit-search-cancel-button {
	display: none; 
}

input[type=search] {
	background: no-repeat 420px center;
	background-color: transparent;
	border: none;
	padding: 12px 10px 9px 9px;
	width: 350px;
	-webkit-transition: all .5s;
	-moz-transition: all .5s;
	transition: all .5s;
}

input[type=search]:focus {
	background: no-repeat 500px center;
	width: 370px;
	border-color: white;
	color: white;

	-webkit-box-shadow: 0 0 5px rgba(109,207,246,.5);
	-moz-box-shadow: 0 0 5px rgba(109,207,246,.5);
	box-shadow: 0 0 5px rgba(109,207,246,.5);
}

input:-moz-placeholder {
	color: white;
}

input::-webkit-input-placeholder {
	font-size: 23px;
	color: white;
}

@media (max-width: 960px){
	input[type=search]:focus {
		width: 200px;
	}

	input[type=search] {
		width: 150px;
	}

	.logo{
		margin-left: 10px;
	}

	.form_search {
		margin-right: 10px; 
	}
}

@media (max-width: 480px){
	.three1 {
		display: block;
	}

	input[type=search] {
		width: 72%;
	}

	input[type=search]:focus {
		width: 72%;
	}

	.form_search {
		margin-right: 1px;
	}
	.search_wrapper {
		margin: 20px 3px 0px 3px;
		font-size: 12px;
	}
}



