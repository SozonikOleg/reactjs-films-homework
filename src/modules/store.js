import { createStore } from 'redux';

function counter(
  state = {
    oleg: 'First date',
    inna: 'Second date',
  },
  action,
) {
  switch (action.type) {
    case 'INCREMENT':
      return state + 1;
    case 'DECREMENT':
      return state - 1;
    default:
      return state;
  }
}

const store = createStore(counter);

export default store;
