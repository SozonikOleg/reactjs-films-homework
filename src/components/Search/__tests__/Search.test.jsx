import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Search from '../Search';
import store from '../../../modules/store';

test('Link changes the class when hovered', () => {
  const result = renderer.create(
    <Provider store={store}>
      <Search />
    </Provider>,
  );
  expect(result).toMatchSnapshot();
});
