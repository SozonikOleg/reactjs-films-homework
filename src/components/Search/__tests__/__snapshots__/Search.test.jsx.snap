// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="three1"
>
  <section
    className="logo"
  >
    <h2>
       Films
    </h2>
  </section>
  <section
    className="form_search"
  >
    <form
      id="myForm"
    >
      <div
        className="search_wrapper"
      >
        <input
          onChange={[Function]}
          placeholder="the jungle book"
          type="search"
          value=""
        />
        <a
          href="/?search="
          onClick={[Function]}
        >
          <button
            className="search_submit"
            onClick={[Function]}
            type="submit"
          />
        </a>
      </div>
    </form>
  </section>
</div>
`;
